Adding fcls for CAFMaker

pgranger23 opened the pull request:
Adding several fcls files to run the FD CAFMaker module.
Energy and angle reconstruction are recomputed to allow for an easy integration of reco improvements.

absolution1 reviewed (commented):
Hi Pierre.

Thank you for prepping the PR.

A couple of Qs.
 - I think the energy reco is included as part of the reconstrcution for beam.  Does the CAF job run over a beam reco2 file?  Does this energy reco config run anything differently?
  - Isn't the angular reco run for the atmospheric reco2 already?

pgranger23 commented:
Hi Dominic,
Indeed, these algorithms are already run as part of reco2. However, they have been subject to several improvements lately, and I felt like re-running them right before producing the CAFs would allow for the integration of the latest modifications (moreover, the processing time of these steps is relatively small).
I guess this might lead to some "reproducibility" issues, but at least on the atmospherics side, I think this is relevant to rerun these algorithms during the CAF production.

absolution1 commented:
Hi Pierre.
OK I see what you mean.  Just to double check, the energy reco has had one change since the last production run, right?
 - https://github.com/DUNE/dunereco/pull/101
 or are there others?
 
If the only change to the energy reco is to force MCS or ranging then I agree with rerunning.  However, I don't think that the standard fcls should rerun the energy/angular reco.  I know this is a pain, but could you add in one more set of fcls that inherits from the standard fcls and also reruns the energy/angular reco (and take those reprocessings out of the standard fcls)?

absolution1 commented:
> Ok. So for now I think this is the only change. However Henrique has made some improvements lately in the energy reco that he expects to be able to present soon. Sure, additional fcls seems to be a solution to fulfill all needs, I'll try to add them ASAP. Do you have any naming suggestions for those?

Sounds good.  For the naming, something like: cafmaker_atmos_dune10kt_1x2x6_runreco-nuenergy-nuangular.fcl
No option is going to be particularly pretty but I think the main things are to 1) drop 'standard' 2) some kind of identifier at the end of the fcl

pgranger23 commented:
Should be done by now. I tested part of the fcls and they seem correct.

absolution1 commented on fcl/dunefd/caf/cafmaker_dune10kt_1x2x6_runreco-nuenergy.fcl:
```diff
@@ -0,0 +1,66 @@
+physics.trigger_paths: [ prod ] 
+
+
+physics.producers.energyrecnumu.HitLabel: "hitfd"
+physics.producers.energyrecnumu.WireLabel: "tpcrawdecoder:gauss"
+physics.producers.energyrecnumu.HitToSpacePointLabel: "hitfd"
+
+physics.producers.energyrecnumurange.HitLabel: "hitfd"
+physics.producers.energyrecnumurange.WireLabel: "tpcrawdecoder:gauss"
+physics.producers.energyrecnumurange.HitToSpacePointLabel: "hitfd"
+
+physics.producers.energyrecnumumcs.HitLabel: "hitfd"
+physics.producers.energyrecnumumcs.WireLabel: "tpcrawdecoder:gauss"
+physics.producers.energyrecnumumcs.HitToSpacePointLabel: "hitfd"
+
+physics.producers.energyrecnue.HitLabel: "hitfd"
+physics.producers.energyrecnue.WireLabel: "tpcrawdecoder:gauss"
+physics.producers.energyrecnue.HitToSpacePointLabel: "hitfd"
+
+physics.producers.energyrecnc.HitLabel: "hitfd"
+physics.producers.energyrecnc.WireLabel: "tpcrawdecoder:gauss"
+physics.producers.energyrecnc.HitToSpacePointLabel: "hitfd"
```
Do these need setting here?  The upstream fcl blocks e.g. `dunefd_nuenergyreco_pandora_numu` don't already set the variables correctly?

pgranger23 replied:
Apparently no, they are set differently in energyreco.fcl -> https://github.com/DUNE/dunereco/blob/fb5a5d2acef112785e3078c8d23e2c617114518c/dunereco/FDSensOpt/energyreco.fcl#L34

I admit this might be better to modify it directly there, I can take care of this.

absolution1 commented on fcl/dunefdvd/caf/cafmaker_dunevd10kt_1x8x6_3view_30deg_runreco-nuenergy.fcl:
```diff
@@ -0,0 +1,28 @@
+    @table::dunefdvd_1x8x6_3view_30deg_reco_services
+}
+
+physics.producers.energyrecnumu.HitLabel: "gaushit"
+physics.producers.energyrecnumu.WireLabel: "tpcrawdecoder:gauss"
+physics.producers.energyrecnumu.HitToSpacePointLabel: "pandora"
+
+physics.producers.energyrecnumurange.HitLabel: "gaushit"
+physics.producers.energyrecnumurange.WireLabel: "tpcrawdecoder:gauss"
+physics.producers.energyrecnumurange.HitToSpacePointLabel: "pandora"
+
+physics.producers.energyrecnumumcs.HitLabel: "gaushit"
+physics.producers.energyrecnumumcs.WireLabel: "tpcrawdecoder:gauss"
+physics.producers.energyrecnumumcs.HitToSpacePointLabel: "pandora"
+
+physics.producers.energyrecnue.HitLabel: "gaushit"
+physics.producers.energyrecnue.WireLabel: "tpcrawdecoder:gauss"
+physics.producers.energyrecnue.HitToSpacePointLabel: "pandora"
+
+physics.producers.energyrecnc.HitLabel: "gaushit"
+physics.producers.energyrecnc.WireLabel: "tpcrawdecoder:gauss"
+physics.producers.energyrecnc.HitToSpacePointLabel: "pandora"
```
same question as above

absolution1 commented on fcl/dunefd/caf/standard_cafmaker_dune10kt_1x2x6.fcl:
```diff
@@ -0,0 +1,46 @@
+  MemoryTracker:         {}
+  message:      @local::dune_message_services_prod_debug #TODO: Change for prod
+  FileCatalogMetadata:  @local::art_file_catalog_mc
+  @table::dunefd_simulation_services
```
This should use the 1x2x6 service block.
Should this use the reco services or the simulation services?

pgranger23 replied:
Do you have a specific example? I admit I never really know which of those have to be used in which circumstances.

pgranger23 replied:
`@table::dunefd_1x2x6_reco_services` I guess

absolution1 commented on fcl/dunefdvd/caf/standard_cafmaker_dunevd10kt_1x8x6_3view_30deg.fcl:
```diff
@@ -0,0 +1,8 @@
+#include "standard_cafmaker_dune10kt_1x2x6.fcl"
```
We've been avoiding having the VD fcls inherit from the HD fcls.  Could you copy the contents of the HD fcl here (but swap out the service block to VD 1x8x6 3view 3deg)?

absolution1 commented on fcl/dunefdvd/caf/standard_cafmaker_dunevd10kt_1x8x6_3view_30deg.fcl:
```diff
@@ -0,0 +1,8 @@
+
+services: {
+    @table::services
+    @table::dunefdvd_reco_services
+    @table::dunefdvd_1x8x6_3view_30deg_reco_services
```
Only the 30deg service block is needed out of these two

absolution1 reviewed (commented):
Thanks for making the changes Pierre, I have one last set of comments.

pgranger23 commented:
I think (hope) I addressed all the comments. I moved the energyreco config in https://github.com/DUNE/dunereco/pull/107 , and this PR now depends on this dunereco one.

tomjunk commented:
I merged the dunereco one and will get this one in this week's release.

tomjunk reviewed (approved):
Looks okay
